enzyme/Enzyme/AdjointGenerator.h

```diff
@@ -145,9 +145,10 @@ class AdjointGenerator : public llvm::InstVisitor<AdjointGenerator> {
         C = CE->getOperand(0);
       }
       if (auto GV = dyn_cast<GlobalVariable>(C)) {
-        if (GV->getName() == "ompi_mpi_double") {
+        auto name = GV->getName();
```

Comment:
add test cases for these

Reply:
Can you provide me a link for where we read outside ptr values (and if it's the right approach)?
otherwise the testcase won't pass, see my error description above.

Reply:
Done. I'll update it or add a second one once we found a solution for the reduction operator.